groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -985,12 +985,16 @@ struct Optional_DataImp {
     TYPE&& value() &&;
         // Return the 'value_type' object in 'd_buffer' with const
         // qualification adjusted to match that of 'TYPE'.  The behavior is
-        // undefined unless 'this->hasValue() == true'
+        // undefined unless 'this->hasValue() == true'.  We do not assert in
+        // this function as the assert level is determined by the 'optional'
+        // method invoking 'value()'
```

Comment:
Assertions are never part of function contract.  I suggest putting these comments in the function implementation instead.

Reply:
moved the assertion comments to the implementation.